### PR TITLE
Add the support of custom user client cert generation

### PR DESCRIPTION
**Pull request by prafull01**
User can generate client certificates for any user using following job:
```
apiVersion: batch/v1
kind: Job
metadata:
  name: user-cert-generate
spec:
  backoffLimit: 1
  completions: 1
  parallelism: 1
  template:
    spec:
      restartPolicy: Never
      containers:
      - args:
        - generate
        - --client-only
        env:
        - name: USER_NAME
          value: prafull              ## Provide the user name to which you want to create client certificates
        - name: CA_SECRET
          value: crdb-cockroachdb-ca-secret  ## CA secret where the CA is stored for CRDB
        - name: NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
        image: gcr.io/cockroachlabs-helm-charts/cockroach-self-signer-cert:dc73bb7f9a40013696fa342af78872fdaa927163
        imagePullPolicy: Always
        name: user-cert-job
      serviceAccountName: crdb-cockroachdb-self-signer
```

### CODE REVIEW

File: pkg/generator/generate_cert.go [hunk 1]
```diff
@@ -145,32 +146,47 @@ func (rc *GenerateCert) Do(ctx context.Context, namespace string) error {
 	return nil
 }
 
+func (rc *GenerateCert) ClientCertGenerate(ctx context.Context, namespace string) error {
```
Comment (keith-mcclellan) [inline pkg/generator/generate_cert.go:149]: Can I get some comments here please

File: pkg/generator/generate_cert.go [hunk 2]
```diff
@@ -517,3 +542,25 @@ func (rc *GenerateCert) UpdateNewCA(ctx context.Context, namespace string) error
 	}
 	return nil
 }
+
+func (rc *GenerateCert) LoadCASecret(ctx context.Context, namespace string) error {
```
Comment (keith-mcclellan) [inline pkg/generator/generate_cert.go:546]: Please put a code comment here as well

**Review by keith-mcclellan — CHANGES_REQUESTED** [review]
Please add some additional comments for the new functions, otherwise LGTM

File: pkg/generator/generate_cert.go (hunk 1 above)
Comment (prafull01) [reply pkg/generator/generate_cert.go:150]: done

File: pkg/generator/generate_cert.go (hunk 2 above)
Comment (prafull01) [reply pkg/generator/generate_cert.go:548]: done

**Review by keith-mcclellan — APPROVED** [review]

**Pull request merged**